### PR TITLE
Makefile: add alembic-new-migration target

### PR DESCRIPTION
**Pull request by jgehrcke**
This is for https://github.com/conbench/conbench/issues/519.

### CODE REVIEW

File: conbench/db.py [hunk 1]
```diff
@@ -39,6 +40,7 @@ def configure_engine(url):
         # https://github.com/conbench/conbench/pull/690
         connect_args={"options": "-c timezone=utc -c statement_timeout=60s"},
     )
+    log.info("bind engine to session")
```
Comment (jgehrcke) [inline conbench/db.py:43]: When developing this target I needed to see where things hang and why; this helped.

**Comment by jgehrcke** [comment]
OK, this is now also covered by CI. Last few lines of output from the GHA job:
```
INFO  [alembic.runtime.migration] Will assume transactional DDL.
Generating /app/migrations/versions/e242e362b7d6_dummy_migr_name.py ...  done
a new file was generated in migrations/versions -- it is probably root-owned
run: sudo chown $USER:$(id -gn) migrations/versions/xxx.py
"""dummy_migr_name

Revision ID: e242e362b7d6
Revises: fe3b1041ed15
Create Date: 2023-03-03 15:19:15.660661

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'e242e362b7d6'
down_revision = 'fe3b1041ed15'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    pass
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    pass
    # ### end Alembic commands ###
```

File: Makefile [hunk 2]
```diff
@@ -76,6 +76,22 @@ rebuild-expected-api-docs: run-app-bg
...
+		echo "env var ALEMBIC_MIGRATION_NAME must be set to an expressive name"; \
+		exit 1; \
+	fi
+	docker compose run --detach db
+	CREATE_ALL_TABLES=false docker compose \
+		run -e CREATE_ALL_TABLES app alembic upgrade head
+	CREATE_ALL_TABLES=false docker compose -f docker-compose.yml -f docker-compose.dev.yml \
+		run -e CREATE_ALL_TABLES  app alembic revision --autogenerate -m ${ALEMBIC_MIGRATION_NAME}
+	@echo "a new file was generated in migrations/versions -- it is probably root-owned"
+	@echo 'run: sudo chown $$USER:$$(id -gn) migrations/versions/xxx.py'
```
Comment (jonkeane) [inline Makefile:92]: 🙃 Love cross-filesystem docker permissions issues

**Review by jonkeane — APPROVED** [review]
I'm good with this, thanks. 

Slightly curious if this was bumping into something else you're working on, or a quick win? Either is fine, of course — but would be curious what this is unlocking 😄

**Comment by jgehrcke** [comment]
> Slightly curious if this was bumping into something else you're working on, or a quick win?

When working on https://github.com/conbench/conbench/pull/788 I needed to do a database migration. It was a bit tricky to figure out how to do that. When I have to do that again in the future, I will be happy about the Makefile target.

The docker-compose setup makes it tricky to do the first time, but then allows everyone to do this reliably on their system.

**Pull request merged**